project/FinaglePlugin.scala

```diff
@@ -0,0 +1,31 @@
+import sbt._, Keys._
+
+import com.typesafe.tools.mima.plugin.MimaKeys._
+import com.typesafe.tools.mima.plugin.SbtMima
+import org.typelevel.sbt.NoPublishPlugin
+
+object FinaglePlugin extends AutoPlugin {
```

Comment:
For every finagle module you want to check bincompat of, this plugin creates a synthetic (no-publish) project for it in your build. Each of these projects is configured for MiMa, except we set `mimaCurrentClassfiles` to the current finagle dependency (rather than the project's compile output, which is irrelevant).

Reply:
Then, all of these projects are aggregated into `rootFinagle`.